pkg/controller/installation/installation_controller.go

```diff
@@ -153,22 +185,32 @@ type ReconcileInstallation struct {
 // and what is in the Installation.Spec
 func (r *ReconcileInstallation) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	installation := &integreatlyv1alpha1.Installation{}
-	err := r.client.Get(r.context, request.NamespacedName, installation)
-
+	err := r.client.Get(context.TODO(), request.NamespacedName, installation)
 	if err != nil {
 		if errors.IsNotFound(err) {
 			return reconcile.Result{}, nil
 		}
 		return reconcile.Result{}, err
 	}
 
+	retryRequeue := reconcile.Result{
+		Requeue:      true,
+		RequeueAfter: 10 * time.Second,
+	}
+
+	//context is cancelled on delete of an installation, so if context is cancelled but there is no deletion timestamp
+	//the installation must be created after one was deleted, so recreate a context to use for the new installation.
```

Comment:
It might be because it's late at night, but this context stuff sounds scary. :grimacing: I wonder how much it's really needed (I'm sure I'll look at this tomorrow and say "oh yeah, that's really important actually" :smile: ).

Reply:
yeah, it solves a pretty serious bug we were having, where the operator would be running 2 concurrent reconciles, one regular, and one on a deleted CR. The deleted reconcile would whizz through and delete everything, then the other one would follow after it (with the CR in memory) recreating everything.

This context allows us to prevent that from happening, but it was also stopping the operator from creating again when a new CR was created after the uninstall was finished, this should correct that behaviour.